streaming-payments.md

```diff
@@ -0,0 +1,375 @@
+  account: 'rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+plugin.connect().then(function () {
+  plugin.on('incoming_prepare', function (transfer) {
+    const ilpPacketContents = IlpPacket.deserializeIlpPayment(Buffer.from(transfer.ilp, 'base64'))
+    const parts = ilpPacketContents.account.split('.')
+    // 0: test, 1: crypto, 2: xrp, 3: rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW, 4: userId, 5: paymentId
+    if (parts.length < 6 || typeof users[parts[4]] === 'undefined' || ilpPacketContents.amount !== transfer.amount) {
+      plugin.rejectIncomingTransfer(transfer.id, {}).catch(function () {})
+    } else {
+      const { secret, res } = users[parts[4]]
+      const fulfillment = hmac(secret, ilpPacketContents.account)
+      const condition = hash(fulfillment)
+      if (transfer.executionCondition === base64(condition)) {
+        plugin.fulfillCondition(transfer.id, base64(fulfillment)).then(function () {
+          const letter = ('ABCDEFGHIJKLMNOPQRSTUVWXYZ').split('')[(Math.floor(Math.random() * 26))]
+          res.write(letter)
+        }, function (err) {
+          console.error(err.message)
+        })
+      } else {
+        console.log('no match!', { secret, fulfillment, condition, transfer })
+      }
+    }
+  })
+
+  http.createServer(function (req, res) {
+    const secret = crypto.randomBytes(32)
+    const user = base64(crypto.randomBytes(8))
+    users[user] = { secret, res }
+    console.log('user! writing head', user)
+    res.writeHead(200, {
+      'Pay': [ 1, plugin.getAccount() + '.' + user, base64(secret) ].join(' ')
+    })
+    // Flush the headers in a first TCP packet:
+    res.socket.write(res._header)
+    res._headerSent = true
+  }).listen(8000)
+})
+```
+
+Starting with the last part, `http.createServer`, you can see the flow of the http server
+is a bit simpler; when a request comes in it sends headers, and
+then the body will be sent letter-by-letter, as payments come in. The `Pay` header contains 3 parts:
+
+* amount (in this case, the price of one letter, in XRP)
+* user-specific destination address
+* a Base64-encoded conditionSeed
+
+Note how we are appending `'.' + user` to the shop's Interledger address! This is a special feature of Interledger
+addresses, they can be subnetted endlessly, just add another `.` at the end to convert an account address
+to a ledger prefix, and then add new sub-accounts after that. In this case, we want to know which user is paying
+for letters, so by telling each user a different Interledger sub-address, we can neatly keep our users apart.
+
+When a transfer comes in, the server opens the [ILP packet](https://github.com/interledgerjs/ilp-packet).
+In there is a destination account and a destination amount, which will usually (if all went well) be equal
+to the amount of the transfer. In later tutorials, we will see how one ILP payment can be chained together
```

Comment:
I think it makes more sense to say that multiple transfers are chained together to create one ILP payment, rather than the ILP payment being chained